traffic_ops/traffic_ops_golang/cdn/cdns.go

```diff
@@ -44,31 +47,53 @@ func GetRefType() *TOCDN {
 }
 
 //Implementation of the Identifier, Validator interface functions
-func (cdn *TOCDN) GetID() (int, bool) {
+func (cdn TOCDN) GetID() (int, bool) {
 	return cdn.ID, true
 }
 
-func (cdn *TOCDN) GetAuditName() string {
+func (cdn TOCDN) GetAuditName() string {
 	return cdn.Name
 }
 
-func (cdn *TOCDN) GetType() string {
+func (cdn TOCDN) GetType() string {
 	return "cdn"
 }
 
 func (cdn *TOCDN) SetID(i int) {
 	cdn.ID = i
 }
 
-func (cdn *TOCDN) Validate(db *sqlx.DB) []error {
-	errs := []error{}
-	if len(cdn.Name) < 1 {
-		errs = append(errs, errors.New(`CDN 'name' is required.`))
+func isValidCDNchar(r rune) bool {
```

Comment:
Should this be in the `tovalidate` package? https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/tovalidate/rules.go#L20

Reply:
seemed too specific to the CDN -- would any other likely use it?   if there's an identical rule in another package,  I'd move it..